lib/serialization/examples/custom_serialization.nit

```diff
@@ -0,0 +1,140 @@
+# Example of an ad hoc serializer that is tailored to transform business specific objects into customized representation.
+#
+# In the following, we expose 3 business classes, `E`, `A` and `B`, and a specific business serializer `RestrictedSerializer`.
+# The principle is that the custom serialization logic in enclosed into the `RestrictedSerializer` and that the
+# standard serializer is unchanged.
+#
+# The additional behaviors exposed are:
+#
+# * replace a full business object. (see `E::id`).
+#   instead of serializing an attribute, the custom serializer use a different representation
+# * inject a phantom attribute (see `E::css_class`).
+#   when serializing, the custom serializer inject new attributes.
+# * hide a normally serialized attribute (see `semantic`)
+#   when serializing, the custom serializer hides some specific attributes.
+#
+# The advantage of the approach is that it is done programmatically so can be adapted to real complex use cases.
+# This also require no additional support from the nit engines apart from the already existing serialization infrastructure.
```

Comment:
I'm not sure the last line is useful to the average developer, it looks like a PR description. You could instead compare this approach to the annotations, well only to `noserialize` when used to hide a single attribute. It may help the reader understand the options and the difference.